### DISCUSSION TITLE
Re-flashing the Emporia Backup

### DISCUSSION THREAD

**Discussion by Magfire** [emporia-vue-local/esphome#18] [Q&A] [answered]
Has anyone tried flashing the Emporia factory firmware backup back onto the device after putting ESPHome on it? I was curious and tried it, but I was getting this error: 

"Unexpected error: The firmware binary is invalid (magic byte=FF, should be E9)"

I was using ESPHome Flasher and I got the same error for all the separate factory firmware backup .bin files I had made. These backups were made using the "esptool.py -b 921600 read_flash 0 0x800000 flash_contents.bin" command from the guide. Does any one have any insight on whats going on and how it might be resolved?

**Comment by flaviut** [comment, marked as answer]
I've never used ESPHome flasher, but what about `esptool.py write_flash 0x0 my_app-0x01000.bin`?

How's that work?

**Reply by Magfire** [reply]
That seems to have worked! Thanks.
